eng/pipelines/typespec-ci.yml

```diff
@@ -0,0 +1,54 @@
+jobs:
+- job: TypeSpecCI
+  pool:
+    name: azsdk-pool-mms-ubuntu-2204-general
+    vmImage: ubuntu-22.04
+
+  steps:
+  - script: npm ci
+    displayName: npm ci
+  
+  - template: templates/steps/typespec-ci.yml
+    parameters:
+      Folder: specification/contosowidgetmanager/Contoso.WidgetManager
```

Comment:
I know you are hardcoding for now, but we should definitely do a dynamic discovery of the projects eventually.

Reply:
Yes, I would like to discover projects dynamically, but I also think it's very nice to have one step per project, so you can easily see in the steps list which projects are failing.  So I will need to dynamically generate a matrix which is a little tricky (but Ben knows how).

Reply:
I don't think I would do a dynamic matrix as that would run these all on their own agent which is probably not efficient. We would need to think more about how to run them in their own step from a dynamic list.

Reply:
Yeah a matrix might not work here.  We can validate all the specs dynamically in a single script, but we'll need to make sure it's easy to see a summary of the results (which specs passed and failed), and carefully handle error propagation.